packages/core/core/src/services/server/admin-api.ts

```diff
@@ -3,7 +3,11 @@ import { createAPI } from './api';
 
 const createAdminAPI = (strapi: Core.Strapi) => {
   const opts = {
-    prefix: '', // '/admin';
+    // prefix is the base URL of the server api
+    // it's relative to the server host
+    // if we run the server as / (root) and admin panel on /admin it should be / (empty string)
+    // if we run the server as subpath like /strapi and admin panel as a subpath like /strapi/admin it should be /strapi
+    prefix: strapi.config.get<string>('admin.path').replace(/\/[^\/]+\/?$/, ''), // remove last part of the path
```

Comment:
Not so sure about that though.

Shouldn't this be the responsibility of the nginx (& co) to strip the `/strapi` (sub domain) part from requests & keep the `server.url` pointing to the API root endpoint?

Reply:
@Convly this is similar issue to the one I answered above – when running strapi instance as subpath of main domain, the admin panel will be sending requests to admin api to correct place but admin api itself will not be bonded to correct url, so using `admin.path` configuration here we can bind the admin api to correct address mitigating the issue (ie. admin api should listen `/strapi` instead of `/`)

Reply:
hi @wladyslawczyzewski we never allowed doing this in the past so that the proxy servers were responsible for url rewriting (basically removing the /strapi in your example) from the calls. this allowed for more flexibility in v4 and worked fine. Could you start with just the changes that make it work like in v4 I would like to consider the consequences of manually changing the server prefixes 🙏

Reply:
hi @alexandrebodin I'm not familiar with strapi v4 implementation details but I don't think that reverting this change is possible as all these three issues are tightly coupled to run the strapi with subpath configuration. I will do more investigation early next week as currently I don't have time to work on this and will get back to you.

in the meantime if you get thoughts or will be able to do the investigation by yourself, please post the update here so we will be on the same page. thanks in advance!